Source/Core/Core/HW/WiimoteEmu/WiimoteEmu.cpp

```diff
@@ -704,14 +732,14 @@ void Wiimote::StepDynamics()
   EmulateTilt(&m_tilt_state, m_tilt, 1.f / ::Wiimote::UPDATE_FREQ);
   EmulateCursor(&m_cursor_state, m_ir, 1.f / ::Wiimote::UPDATE_FREQ);
   EmulateShake(&m_shake_state, m_shake, 1.f / ::Wiimote::UPDATE_FREQ);
+  EmulateIMUCursor(&m_imu_cursor_state, m_imu_ir, m_imu_accelerometer, m_imu_gyro,
+                   1.f / ::Wiimote::UPDATE_FREQ);
 }
 
-Common::Vec3 Wiimote::GetAcceleration()
+Common::Vec3 Wiimote::GetAcceleration(Common::Vec3 extra_acceleration)
 {
-  Common::Vec3 accel =
-      GetOrientation() *
-      GetTransformation().Transform(
-          m_swing_state.acceleration + Common::Vec3(0, 0, float(GRAVITY_ACCELERATION)), 0);
+  Common::Vec3 accel = GetOrientation() * GetTransformation().Transform(
```

Comment:
It looks this will rotate "raw" acceleration (e.g. from a DS4) based on the would-be cursor transformation. That should not be done. Acceleration and and rotational-velocity should be passed-through other than `GetOrientation()` rotations caused by Sideways/Upright settings.

Reply:
Well.. I guess GetTransformation is just going to return the emulated transform. Maybe this does make sense.

Reply:
My problem is that I'm not really able to follow the maths. A little bit of understanding and a lot of trial and error is unfortunately my way when heavy (for me) maths is involved.

Reply:
> Instead of working in the "extra" data in GetTransformation and GetAngularVelocity perhaps that should be handled in your GetTotal_ functions where the availability of the raw data is known.

This was my initial idea and attempt. I was however unable to make any headway. I remembered that the order of matrix multiplications is significant, so I tried one by one putting the accelerometer, gyro and cursor values into what seemed like logical locations in the existing calculations. Reasonably quickly I found something that seemed to work with the test programs I use, and later in games.

Unfortunately I don't see myself being able to improve much on the current calculations. However, I will of course be happy to accept changes from others.

> Well.. I guess GetTransformation here is just going to return the emulated transform. Maybe this does make sense.

The only thing I can say is that it seems to work well in practice.

Reply:
I think it is correct as-is. I was incorrectly thinking `GetTransformation()` would include the tracked transformation of the real device.

Reply:
> Well.. I guess GetTransformation here is just going to return the emulated transform.

Wow, my brain just didn't understand this sentence earlier, but now it's clear as day, That's chronic illness for you (or me in this case) 😖

I of course knew that the original `GetAcceleration`, `GetAngularVelocity `and `GetTransformation` only deal with emulated motion.